src/storage-client/src/controller.rs

```diff
@@ -886,25 +890,56 @@ where
                 .await
                 .unwrap();
 
-            let (write, mut read) = persist_client
-                .open(metadata.data_shard)
+            let write = persist_client
+                .open_writer(metadata.data_shard)
                 .await
                 .expect("invalid persist usage");
 
-            // Advance the collection's `since` as requested.
-            if let Some(since) = &description.since {
-                read.downgrade_since(since).await;
-            }
+            // Construct the handle in a separate block to ensure all error paths are diverging
+            let since_handle = {
+                let mut handle = persist_client
+                    .open_critical_since(
+                        metadata.data_shard,
+                        PersistClient::CONTROLLER_CRITICAL_SINCE,
+                    )
+                    .await
+                    .expect("invalid persist usage");
+
+                let since = description
+                    .since
+                    .clone()
+                    .unwrap_or_else(|| handle.since().clone());
+
+                // We should only continue if we can fence out any other processes
+                let our_epoch = self.state.envd_epoch + 1;
```

Comment:
this +1 is a bit weird, i know we can't use the epoch itself because we need to succeed a `compare_and_downgrade_since` against the default epoch of the first since handle, but explaining the +1 here might be nice

Reply:
> we need to succeed a compare_and_downgrade_since against the default epoch of the first since handle

The default epoch is 0 right? The envd epoch function will current never return anything < 1, so I think it is ok to not add one? (This is currently not a documented guarantee, but it can be.)